server/src/main/java/org/opensearch/action/admin/indices/scale/searchonly/ScaleIndexAction.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.indices.scale.searchonly;
+
+import org.opensearch.action.ActionType;
+import org.opensearch.action.support.clustermanager.AcknowledgedResponse;
+
+/**
+ * Action type for search-only scale operations on indices.
+ * <p>
+ * This action type represents the administrative operation to transition indices
+ * between normal operation and search-only mode. It provides the action name constant
+ * used for transport actions and request routing. The action produces an
+ * {@link AcknowledgedResponse} to indicate success or failure.
+ * <p>
+ * The action is registered with the name "indices:admin/scale/search_only" in the
+ * transport action registry.
+ */
+public class ScaleIndexAction extends ActionType<AcknowledgedResponse> {
```

Comment:
I'm not a big fan of this action name. 

It feels too generic for the case that it's trying to address. I'm worried that if we go for the specific name, we'll end up in a similar situation to our existing "search" node type, which really means "searchable snapshot".

That said, the API part of it is the `NAME` field, which does include `search_only`, which is good. For the class name, we have some flexibility to change in the future.

Reply:
Thanks @msfroh thats a good point, I though about it and hence updated the `NAME` and package to include `searchonly`.  So the `ScaleIndexAction` is inside the searchonly package.